src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3466,43 +3467,24 @@ export class StudyViewPageStore {
         const cancerTypeIds = _.uniq(
             this.queriedPhysicalStudies.result.map(study => study.cancerTypeId)
         );
+        const survivalUniqueKeys = this.survivalPlotKeys;
 
-        this.chartsType.set(UniqueKey.OVERALL_SURVIVAL, ChartTypeEnum.SURVIVAL);
-        this.chartsDimension.set(
-            UniqueKey.OVERALL_SURVIVAL,
-            STUDY_VIEW_CONFIG.layout.dimensions[ChartTypeEnum.SURVIVAL]
-        );
-        if (
-            osStatusFlag &&
-            osMonthsFlag &&
-            getDefaultPriorityByUniqueKey(UniqueKey.OVERALL_SURVIVAL) !== 0
-        ) {
-            // hide OVERALL_SURVIVAL chart if cancer type is mixed or have more than one cancer type
-            if (cancerTypeIds.length === 1 && cancerTypeIds[0] !== 'mixed') {
-                this.changeChartVisibility(UniqueKey.OVERALL_SURVIVAL, true);
-            }
-        }
-        this.chartsType.set(
-            UniqueKey.DISEASE_FREE_SURVIVAL,
-            ChartTypeEnum.SURVIVAL
-        );
-        this.chartsDimension.set(
-            UniqueKey.DISEASE_FREE_SURVIVAL,
-            STUDY_VIEW_CONFIG.layout.dimensions[ChartTypeEnum.SURVIVAL]
-        );
-        if (
-            dfsStatusFlag &&
-            dfsMonthsFlag &&
-            getDefaultPriorityByUniqueKey(UniqueKey.DISEASE_FREE_SURVIVAL) !== 0
-        ) {
-            // hide DISEASE_FREE_SURVIVAL chart if cancer type is mixed or have more than one cancer type
-            if (cancerTypeIds.length === 1 && cancerTypeIds[0] !== 'mixed') {
-                this.changeChartVisibility(
-                    UniqueKey.DISEASE_FREE_SURVIVAL,
-                    true
-                );
+        survivalUniqueKeys.map(key => {
```

Comment:
this should probably be forEach.  only use map if you are doing something with result of mapping operation